### PR TITLE
feat(payments): add creation endpoint

### PR DESCRIPTION
**Pull request by KLea**

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -77,3 +77,130 @@ export interface PaymentEligibilityPayload extends PaymentPayloadBase {
...
+  | 'scored_no'
+  | 'scored_maybe'
+  | 'scored_yes'
+  | 'in_progress'
+  | 'paid'
+  | 'late'
+  | 'default'
+
+export interface Payment {
+  billing_address: null
```
Comment (olance) [inline src/types.ts:130]: Shouldn't we have a `Address` type and use that here and with `shipping_address` ? ^^

File: src/types.ts [hunk 2]
```diff
@@ -77,3 +77,130 @@ export interface PaymentEligibilityPayload extends PaymentPayloadBase {
...
+  three_d_secure_possible: boolean
+  verified: boolean
+}
+export interface Installment {
+  customer_fee: number
+  due_date: number
+  original_purchase_amount: number
+  purchase_amount: number
+  // TODO complete
+  state: 'pending'
```
Comment (olance) [inline src/types.ts:107]: Would be great to have at least `'pending' | 'paid'`

File: src/types.ts [hunk 3]
```diff
@@ -77,3 +77,130 @@ export interface PaymentEligibilityPayload extends PaymentPayloadBase {
...
+    email: string
+    email_verified: boolean
+    first_name: string
+    id: string
+    is_business: boolean
+    last_name: string
+    phone: string
+    phone_verified: boolean
+    primary_bank_account: null
+  }
```
Comment (olance) [inline src/types.ts:154]: Could you extract that subtype to a `Customer` type?
And probably rename `CustomerData` to `CustomerPayload` and make `Customer` extend it?

File: src/types.ts [hunk 4]
```diff
@@ -77,3 +77,130 @@ export interface PaymentEligibilityPayload extends PaymentPayloadBase {
 export interface PaymentPayload extends PaymentPayloadBase {
   payment: AtLeastOne<PaymentDataProps, 'billing_address' | 'shipping_address'>
 }
+
+
+export type PaymentId = string
```
Comment (olance) [inline src/types.ts:82]: Why?

And why `PaymentId` instead of just `Id` ?

File: src/types.ts [hunk 5]
```diff
@@ -77,3 +77,130 @@ export interface PaymentEligibilityPayload extends PaymentPayloadBase {
...
+      comment: null
+      created: number
+      customer_url: null
+      data: Record<string, unknown>
+      id: string
+      merchant_reference: string
+      merchant_url: string
+      payment: PaymentId
+    }
+  ]
```
Comment (olance) [inline src/types.ts:178]: Same comment as for the `Customer` type (and `OrderData` & co)

File: src/types.ts [hunk 6]
```diff
@@ -77,3 +77,130 @@ export interface PaymentEligibilityPayload extends PaymentPayloadBase {
...
+  origin: string
+  payment_plan: Installment[]
+  // TODO complete
+  preferred_payment_method: 'card'
+  purchase_amount: number
+  refunds: []
```
Comment (olance) [inline src/types.ts:184]: No refunds? 😁

File: src/types.ts [hunk 7]
```diff
@@ -77,3 +77,130 @@ export interface PaymentEligibilityPayload extends PaymentPayloadBase {
...
+    created: number
+    email: string | null
+    first_name: string | null
+    id: string
+    last_name: string | null
+    line1: string
+    line2: string | null
+    phone: string | null
+    postal_code: string | null
+  }
```
Comment (olance) [inline src/types.ts:202]: Use `Address` type

File: src/types.ts [hunk 8]
```diff
@@ -77,3 +77,130 @@ export interface PaymentEligibilityPayload extends PaymentPayloadBase {
...
+  ]
+  origin: string
+  payment_plan: Installment[]
+  // TODO complete
+  preferred_payment_method: 'card'
+  purchase_amount: number
+  refunds: []
+  requirements: Requirement[]
+  return_url: string
+  seller: null
```
Comment (olance) [inline src/types.ts:187]: There's probably an actual type there? ^^

File: src/types.ts [hunk 9]
```diff
@@ -77,3 +77,130 @@ export interface PaymentEligibilityPayload extends PaymentPayloadBase {
...
+  deferred_days: number
+  deferred_months: number
+  id: PaymentId
+  installments_count: number
+  is_customer_kyced: boolean
+  locale: string
+  merchant_id: string
+  merchant_name: string
+  merchant_website: string | null
+  logo_url: string | null
```
Comment (olance) [inline src/types.ts:166]: According to our `Payment` external representation code, this should be:

```suggestion
  logo_url?: string
```

**Review by Taskim — DISMISSED** [review]

**Review by olance — APPROVED** [review]

**Pull request merged**